docs/concepts/articulations.rst

```diff
@@ -3,6 +3,105 @@ Articulations
+See the table below for the number of generalized coordinates for each joint type.
+
+Maximal coordinates describe the configuration of an articulation in terms of the body link positions and velocities.
+Each rigid body has 7 degrees of freedom to describe its position (3D vector) and orientation (XYZW quaternion) in space, see :attr:`newton.State.body_q`,
+and 6 degrees of freedom to describe its velocity (3D vector for linear velocity and 3D vector for angular velocity) in space, see :attr:`newton.State.body_qd`.
+
+To convert between these two representations we use forward and inverse kinematics:
+forward kinematics (:func:`newton.eval_fk`) converts generalized coordinates to maximal coordinates, and inverse kinematics (:func:`newton.eval_ik`) converts maximal coordinates to generalized coordinates.
+
+In Newton, we support both parameterizations and it is up to the solver which one to use to read and write the configuration.
+For example, :class:`~newton.solvers.SolverMuJoCo` and :class:`~newton.solvers.SolverFeatherstone` use generalized coordinates, while :class:`~newton.solvers.SolverXPBD` and :class:`~newton.solvers.SolverSemiImplicit` use maximal coordinates.
+
+When declaring an articulation using the :class:`~newton.ModelBuilder`, the rigid body poses (maximal coordinates :attr:`newton.State.body_q`) are initialized by the ``xform`` argument:
+
+.. code-block:: python
+
+  builder = newton.ModelBuilder()
+  tf = wp.transform(wp.vec3(1.0, 2.0, 3.0), wp.quat_from_axis_angle(wp.vec3(0.0, 0.0, 1.0), 0.5 * wp.pi))
+  body = builder.add_body(xform=tf)
+  builder.add_shape_box(body)  # add a shape to the body to add some inertia
+
+  model = builder.finalize()
+  state = model.state()
+
+  # The body poses (maximal coordinates) are initialized by the xform argument:
+  # model.body_q[0] == state.body_q[0] == tf
+
+  # However, the generalized coordinates are empty:
+  # len(state.joint_q) == 0
+
+In this setup, we have a body with a box shape that maximal-coordinate solvers can directly simulate
+given the initial body pose we defined above.
+
+However, to be able to simulate the same scene using a generalized-coordinate solver, we need to add a free joint to connect the body to the world and make sure the system
+has the degrees of freedom in generalized coordinates (:attr:`newton.State.joint_q`) we need:
+
+.. code-block:: python
+
+  builder = newton.ModelBuilder()
+  tf = wp.transform(wp.vec3(1.0, 2.0, 3.0), wp.quat_from_axis_angle(wp.vec3(0.0, 0.0, 1.0), 0.5 * wp.pi))
+  body = builder.add_body(xform=tf)
+  builder.add_shape_box(body)  # add a shape to the body to add some inertia
+  builder.add_joint_free(body)  # add a free joint to connect the body to the world
+  # The free joint's coordinates (joint_q) are initialized by its child body's pose,
+  # so we do not need to specify them here
+  # builder.joint_q[-7:] = *tf
```

Comment:
Why this notation? Because the joint was added last and therefore it'll be the last 7 elements?

Reply:
And like above maybe the simple pendulum is a better example.

But I agree the free joint for floating-base articulations needs a special mention.